src/main/java/org/opensearch/ad/ml/EntityColdStarter.java

```diff
@@ -346,7 +380,12 @@ private void trainModelFromDataSegments(List<double[][]> dataPoints, Entity enti
             // vector is x1, x2, x3, x4, now we add x3, x4, x5, x6. RCF will recognize
             // overlapping x3, x4, and only store x5, x6.
             .shingleSize(shingleSize)
-            .build();
+            .internalShinglingEnabled(true);
```

Comment:
Does this mean we can just feed un-shingled data to RCF and RCF will do shingling inside?

Reply:
yes.

Reply:
Got it, is this disabled by default? Not sure if this will impact current logic as we are feeding shingled data into RCF.

Reply:
I am changing only HCAD real time to have internalShinglingEnabled. We fed unshingled data to old HCAD realtime RCF models.

For HCAD historical, the model is still using shingled data.